tools/src/icon4py/tools/py2fgen/wrappers/dycore_wrapper.py

```diff
@@ -236,11 +238,11 @@ def solve_nh_init(
     )
 
     # datatest config CellParams
-    cell_geometry = grid_states.CellParams.from_global_num_cells(
+    cell_geometry = grid_states.CellParams(
         cell_center_lat=cell_center_lat,
         cell_center_lon=cell_center_lon,
         area=cell_areas,
-        global_num_cells=dycore_wrapper_state["grid"].global_properties.num_cells,
+        mean_cell_area=mean_cell_area,
```

Comment:
@halungge Before this change, mean_cell_area was computed, I guess it's safer to use this approach.

Reply:
yes. ICON does several things for this quantity: if you run with a DWD grid file (icosahedron) mean_cell_area is computed from the R and B parameter of the grid as we do it in icon4py. If you use a MPI-M generated grid, it is computed in their grid_generator and they do a global reduction over the cell_areas for the icosahedron. (If you run on a torus it is analytically computed. ) So you don't know what you get and for limited_area models there might be a difference between the two approaches.